lib/elixir/lib/option_parser.ex

```diff
@@ -394,16 +394,31 @@ defmodule OptionParser do
       ["--bool", "--no-bool"]
 
   """
-  @spec to_argv(Enumerable.t) :: argv
-  def to_argv(enum) do
+  @spec to_argv(Enumerable.t, options) :: argv
+  def to_argv(enum, opts \\ []) do
```

Comment:
Instead of passing the options and calling `compile_config(opts)` for every entry, I think you could, right before this line, invoke: `Keyword.get(opts, :switches, [])` and just pass the switches to `to_argv/3`.